pkg/openapiclient/local_schemas.go

```diff
@@ -14,55 +13,49 @@ import (
 
 // client which provides openapi read from files on disk
 type localSchemasClient struct {
-	fs  fs.FS
-	dir string
+	fs fs.FS
 }
 
 // Dir should have openapi files following directory layout:
 // /<apis>/<group>/<version>.json
 // /api/<version>.json
-func NewLocalSchemaFiles(fs fs.FS, dirPath string) openapi.Client {
+func NewLocalSchemaFiles(fs fs.FS) openapi.Client {
 	return &localSchemasClient{
-		fs:  fs,
-		dir: dirPath,
+		fs: fs,
 	}
 }
 
 func (k *localSchemasClient) Paths() (map[string]openapi.GroupVersion, error) {
-	if len(k.dir) == 0 {
+	if k.fs == nil {
 		return nil, nil
 	}
 	res := map[string]openapi.GroupVersion{}
-	apiGroups, err := utils.ReadDir(k.fs, filepath.Join(k.dir, "apis"))
-	if err != nil && !errors.Is(err, fs.ErrNotExist) {
-		return nil, fmt.Errorf("failed reading local files dir %s: %w", filepath.Join(k.dir, "apis"), err)
-	}
-	for _, f := range apiGroups {
-		groupPath := filepath.Join(k.dir, "apis", f.Name())
-		versions, err := utils.ReadDir(k.fs, groupPath)
-		if err != nil {
-			return nil, fmt.Errorf("failed reading local files dir %s: %w", groupPath, err)
+	if apiGroups, err := fs.ReadDir(k.fs, "apis"); err == nil {
+		for _, f := range apiGroups {
+			groupPath := path.Join("apis", f.Name())
+			versions, err := fs.ReadDir(k.fs, groupPath)
+			if err != nil {
+				return nil, fmt.Errorf("failed reading local files dir %s: %w", groupPath, err)
+			}
+			for _, v := range versions {
+				if !utils.IsJson(v.Name()) {
+					continue
+				}
+				name := strings.TrimSuffix(v.Name(), path.Ext(v.Name()))
+				apisPath := path.Join("apis", f.Name(), name)
+				res[apisPath] = groupversion.NewForFile(k.fs, path.Join(groupPath, v.Name()))
+			}
 		}
-		for _, v := range versions {
+	}
+	if coregroup, err := fs.ReadDir(k.fs, "api"); err == nil {
```

Comment:
It's not `err != nil`, it's using `==` ;-)

Reply:
I am suggesting to switch to `!=` to avoid nesting:

```go
coregroup, err := fs.ReadDir(k.fs, "api")
if err != nil {
    // Ignoring error because ...


    // coregroup == [] in this case, safe to not return here
}
...
```

Reply:
Oh ok, i'll change that.

Reply:
The linter does not like empty if statements though 🤔